src/Raven.Server/Documents/Replication/ShardReplicationLoader.cs

```diff
@@ -169,4 +170,15 @@ private void HandleMigrationReplication(DatabaseRecord newRecord, List<IDisposab
             }
         }
     }
+
+    public static ExternalReplicationState GetShardedExternalReplicationState(ServerStore server, string database, string source, string sourceDbId)
```

Comment:
let's get them all in a single read tx and return here a dictionary

Reply:
done